raiden-network/light-client
# Support customising gasPrice strategy

Fixes #2795 

**Short description**
This allows some control of `gasPrice` with which transactions are sent. On dApp/Metamask, this controls only the default presented `gasPrice` in MM's tx prompt. On CLI, `--gas-price` also can receive some aliases, like `fast` (1.2) or `medium` (1.05). When not set, the default is 1.0. This is a multiplier for the current ETH-provider-returned `eth_gasPrice`.

A better fix, presented on the issue, is to instead be able to even use lower gasPrices by default, but be able to bump them for priority transactions (txs with deadlines, like `register secret` or `channel update`, but this should be good enough for now. Eventually, those strategies could also be exposed through the API and let the user set it per operation.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Check CLI with `--gas-price fast` option makes scenarios send transactions with +20% the current avg `gasPrice`
